apps/docs/transformers/code-block-demo/index.js

```diff
@@ -19,111 +16,90 @@ const CBD_DEMO = 'cbd-demo';
+	preBlocks.forEach((pre, index) => {
+		const code = pre.querySelector(':scope > code');
+		const src = createiFrameContent(code.textContent, pre.classList, index, outputPath);
+		const fragment = renderiFrame(index, src, pre.outerHTML);
+		pre.replaceWith(fragment);
 	});
-	return jsdomObj.serialize();
+	return dom.serialize();
 };
 
-const getHtml = (demoData) => {
-	const codeBlockId = `${CBD_CODE_BLOCK}-${demoData.index}`;
-	const frameData = {};
-	frameData.demoStr = demoData.demoStr;
-	frameData.codeBlockId = codeBlockId;
-	frameData.outputPath = demoData.outputPath;
-	const iframeSrc = getIframe(frameData);
-
-	return `
-    <vwc-card elevation="0" class="${CBD_CONTAINER}">
-      <iframe class="${CBD_DEMO}" src="${iframeSrc}" onload=onloadIframe(this) loading="lazy" aria-label="code block preview iframe" slot="main"></iframe>
-      <vwc-action-group appearance="ghost" style="direction: rtl;" slot="main">
-        <vwc-button aria-label="Show source code" icon="code-line" aria-expanded="false" aria-controls="${codeBlockId}" onclick="codeBlockButtonClick(this)"></vwc-button>
-        <vwc-button aria-label="Copy source code" icon="copy-2-line" onclick="codeCopyButtonClick(this)"></vwc-button>
-      </vwc-action-group>
-      <details class="${CBD_DETAILS}" slot="main">
-        <summary></summary>
-        <div class="${CBD_CODE_BLOCK}" role="region" id="${codeBlockId}">
-          ${demoData.codeStr}
-        </div>
-      </details>
-    </vwc-card>`;
-}
-
-const getIframe = (frameData) => {
-	const saveFolder = verifyAndCreateSaveFolder(frameData.outputPath);
-	frameData.saveFolder = saveFolder;
-	const filePath = saveCodeAsHTMLFile(frameData);
-	return filePath.substring(saveFolder.indexOf('docs' + path.sep) + 4);
-}
+const renderiFrame = (index, src, content) => JSDOM.fragment(`
+	<div style="--tooltip-inline-size: 160px;">
+		<vwc-card elevation="0" class="${CBD_CONTAINER}">
+			<iframe id="iframe-sample-${index}" src="${src}" class="${CBD_DEMO}" onload=onloadIframe(this) loading="lazy" aria-label="code block preview iframe" slot="main"></iframe>
+			<vwc-action-group appearance="ghost" style="direction: rtl;" slot="main">
+				<vwc-button id="buttonEdit${index}" aria-label="Edit source code" icon="code-line" aria-expanded="false" aria-controls="${CBD_CODE_BLOCK}-${index}" onclick="codeBlockButtonClick(this)"></vwc-button>
+				<vwc-button id="buttonCopy${index}" aria-label="Copy source code" icon="copy-2-line" data-index="${index}"></vwc-button>
+			</vwc-action-group>
+			<details class="${CBD_DETAILS}" slot="main">
+				<summary></summary>
+				<div class="cbd-live-sample" data-index="${index}" role="region">
+					${content}
+				</div>
+			</details>
+		</vwc-card>
+		<vwc-tooltip anchor="buttonEdit${index}" text="Edit source code" placement="top-start"></vwc-tooltip>
+		<vwc-tooltip anchor="buttonCopy${index}" text="Copy source code" placement="top-start"></vwc-tooltip>
```

Comment:
```suggestion
		<vwc-tooltip anchor="buttonEdit${index}" text="Edit" placement="top-start"></vwc-tooltip>
		<vwc-tooltip anchor="buttonCopy${index}" text="Copy" placement="top-start"></vwc-tooltip>
```

@olaf-k Maybe just copy and edit to make it smaller? what do you think?

Reply:
let's settle on "Edit code" and "Copy code" as in your example. short and clear.